instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/net/PeerServiceAttributesExtractor.java

```diff
@@ -53,27 +53,44 @@ public void onEnd(
       @Nullable RESPONSE response,
       @Nullable Throwable error) {
 
-    if (peerServiceMapping.isEmpty()) {
+    if (peerServiceResolver.isEmpty()) {
       // optimization for common case
       return;
     }
 
     String serverAddress = attributesGetter.getServerAddress(request);
-    String peerService = mapToPeerService(serverAddress);
+    Integer serverPort = attributesGetter.getServerPort(request);
+    String path = getUrlPath(attributesGetter, request);
+    String peerService = mapToPeerService(serverAddress, serverPort, path);
     if (peerService == null) {
       String serverSocketDomain = attributesGetter.getServerSocketDomain(request, response);
-      peerService = mapToPeerService(serverSocketDomain);
+      Integer serverSocketPort = attributesGetter.getServerSocketPort(request, response);
+      peerService = mapToPeerService(serverSocketDomain, serverSocketPort, null);
     }
     if (peerService != null) {
       attributes.put(SemanticAttributes.PEER_SERVICE, peerService);
     }
   }
 
   @Nullable
-  private String mapToPeerService(@Nullable String endpoint) {
-    if (endpoint == null) {
+  private String mapToPeerService(
+      @Nullable String host, @Nullable Integer port, @Nullable String path) {
+    if (host == null) {
       return null;
     }
-    return peerServiceMapping.get(endpoint);
+    return peerServiceResolver.resolveService(host, port, path);
+  }
+
+  @Nullable
+  private String getUrlPath(
+      ServerAttributesGetter<REQUEST, RESPONSE> attributesGetter, REQUEST request) {
+    if (attributesGetter instanceof HttpClientAttributesGetter<?, ?>) {
+      String urlFull =
+          ((HttpClientAttributesGetter<REQUEST, RESPONSE>) attributesGetter).getUrlFull(request);
```

Comment:
what do you think about having an additional HttpClientPeerServiceAttributesExtractor to be more explicit about this dependency?

Reply:
Done